src/main/java/teammates/ui/webapi/action/UpdateInstructorPrivilegeAction.java

```diff
@@ -0,0 +1,141 @@
+package teammates.ui.webapi.action;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Const;
+import teammates.ui.webapi.output.InstructorPrivilegeData;
+import teammates.ui.webapi.request.InstructorPrivilegeUpdateRequest;
+
+/**
+ * Update instructor privilege by someone with instructor modify permission.
+ */
+public class UpdateInstructorPrivilegeAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.getId());
+
+        gateKeeper.verifyAccessible(
+                instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        String idOfInstructorToUpdate = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_ID);
+        InstructorAttributes instructorToUpdate = logic.getInstructorForGoogleId(courseId, idOfInstructorToUpdate);
```

Comment:
The current way of identifying instructor works (by (courseId, googleId)). However, in the page where instructor's privilege is edited, it is possible to edit the privilege of an unregistered instructor? So it would be better to use (courseId, email) as the pair to identify the instructor to update.